scipy/linalg/basic.py

```diff
@@ -1617,3 +1599,202 @@ def matrix_balance(A, permute=True, scale=True, separate=False,
     iperm[perm] = np.arange(n)
 
     return B, np.diag(scaling)[iperm, :]
+
+
+def _validate_args_for_toeplitz_ops(c_or_cr, b, check_finite, keep_b_shape):
```

Comment:
Checked this. Looks like it does the same thing as before.

Reply:
That's good right? I didn't want to make any changes to that function - just split the code to make it a bit cleaner.

Reply:
Yes. Just recording that I did look at it.